File: leadMangement/src/main/java/com/management/lead/leadmangement/dto/LeadDTO.java
package com.management.lead.leadmangement.dto;

import java.math.BigDecimal;
import java.util.Date;
import jakarta.persistence.Temporal;
import jakarta.persistence.TemporalType;
import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.Digits;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@RequiredArgsConstructor
public class LeadDTO {

    private String quote;

    @DecimalMin(value = "0.00", inclusive = false)
    @Digits(integer = 10, fraction = 2)
    private BigDecimal budget;

    private Double probability;

    private String priority;

    private String stage;

    private String source;

    private String tag;

    @NotNull
    @Temporal(TemporalType.DATE)
    private Date expectedClosingDate;

    private long assignedUser;
}

File: leadmanageclient/src/app/model/LeadDTO.ts
import { UserDTO } from "./UserDTO";

export class LeadDTO {
    constructor(
        public quote: string = '',
        public budget: number = 0,
        public probability: string = '',
        public priority: string = '',
        public stage: string = '',
        public source: string = '',
        public tag: string = '',
        public expectedClosingDate?: Date,
        public assignedUser: UserDTO = new UserDTO()
    ) { }

    reset(): void {
        this.quote = '';
        this.budget = 0;
        this.probability = '';
        this.priority = '';
        this.stage = '';
        this.source = '';
        this.tag = '';
        this.expectedClosingDate = undefined;
        this.assignedUser = new UserDTO();
    }
}
